frontend/src/components/TopicCard/Follow.js

```diff
@@ -3,16 +3,17 @@ import PropTypes from 'prop-types'
 
 class Follow extends Component {
   render = () => {
-    const { isFollowing, onTopicFollow } = this.props
-    return <div className='topicFollow' onClick={onTopicFollow}>
-      {isFollowing ? 'Unfollow' : 'Follow'}
+    const { ActiveMapper, isFollowing, onTopicFollow } = this.props
+    return <div className='topicFollow' onClick={() => ActiveMapper && onTopicFollow()}>
```

Comment:
I don't think it's really a bottleneck for us, so it probably doesn't matter, but we should consider a broader strategy for these types of functions. The best way to do this kind of thing is declare a function in the class using an arrow function so `this` is automatically bound (see https://medium.freecodecamp.org/react-binding-patterns-5-approaches-for-handling-this-92c651b5af56 for a good discussion on the relative merits of different ways)

But this is concise and developer friendly so I don't think you need to change it here. But we should consider maybe someday refactoring our components to follow one pattern or the other, especially if we start thinking about JS performance at any time

Reply:
I think we'd want to set up some profiling to really dive into js perf

Reply:
I don't follow you here, *which* type of functions?

Reply:
Putting an arrow function in a prop; it has go reallocate memory or something for the function each time it renders

Reply:
oh i c i c